Cannot find name 'File' error after v5 upgrade (in Node)

Hi Sanity team

A similar issue [to this one](https://github.com/IBM/node-sdk-core/issues/9) happens after the 5.2.1 update (console below)

In Node if the tsconfig / lib options doesn't include 'dom' option (which can happen if you are using this package on server side)

```
node_modules/.pnpm/@sanity+client@5.2.1/node_modules/@sanity/client/dist/index.d.ts:47:11 - error TS2304: Cannot find name 'File'.

47     body: File | Blob | Buffer | NodeJS.ReadableStream,
             ~~~~

node_modules/.pnpm/@sanity+client@5.2.1/node_modules/@sanity/client/dist/index.d.ts:59:11 - error TS2304: Cannot find name 'File'.

59     body: File | Blob | Buffer | NodeJS.ReadableStream,
             ~~~~

node_modules/.pnpm/@sanity+client@5.2.1/node_modules/@sanity/client/dist/index.d.ts:71:11 - error TS2304: Cannot find name 'File'.

71     body: File | Blob | Buffer | NodeJS.ReadableStream,
             ~~~~

node_modules/.pnpm/@sanity+client@5.2.1/node_modules/@sanity/client/dist/index.d.ts:589:11 - error TS2304: Cannot find name 'File'.

589     body: File | Blob | Buffer | NodeJS.ReadableStream,
              ~~~~

node_modules/.pnpm/@sanity+client@5.2.1/node_modules/@sanity/client/dist/index.d.ts:605:11 - error TS2304: Cannot find name 'File'.

605     body: File | Blob | Buffer | NodeJS.ReadableStream,
              ~~~~

node_modules/.pnpm/@sanity+client@5.2.1/node_modules/@sanity/client/dist/index.d.ts:621:11 - error TS2304: Cannot find name 'File'.

621     body: File | Blob | Buffer | NodeJS.ReadableStream,
```

--- a/src/assets/AssetsClient.ts
+++ b/src/assets/AssetsClient.ts
@@ -10,6 +10,7 @@ import type {
   ResponseEvent,
   SanityAssetDocument,
   SanityImageAssetDocument,
+  UploadBody,
   UploadClientConfig,
 } from '../types'
 import * as validators from '../validators'
@@ -32,7 +33,7 @@ export class ObservableAssetsClient {
    */
   upload(
     assetType: 'file',
-    body: File | Blob | Buffer | NodeJS.ReadableStream,
+    body: UploadBody,
     options?: UploadClientConfig
   ): Observable<HttpRequestEvent<{document: SanityAssetDocument}>>
 
@@ -45,7 +46,7 @@ export class ObservableAssetsClient {
    */
   upload(
     assetType: 'image',
-    body: File | Blob | Buffer | NodeJS.ReadableStream,
+    body: UploadBody,
     options?: UploadClientConfig
   ): Observable<HttpRequestEvent<{document: SanityImageAssetDocument}>>
   /**
@@ -57,12 +58,12 @@ export class ObservableAssetsClient {
    */
   upload(
     assetType: 'file' | 'image',
-    body: File | Blob | Buffer | NodeJS.ReadableStream,
+    body: UploadBody,
     options?: UploadClientConfig
   ): Observable<HttpRequestEvent<{document: SanityAssetDocument | SanityImageAssetDocument}>>
   upload(
     assetType: 'file' | 'image',
-    body: File | Blob | Buffer | NodeJS.ReadableStream,
+    body: UploadBody,
     options?: UploadClientConfig
   ): Observable<HttpRequestEvent<{document: SanityAssetDocument | SanityImageAssetDocument}>> {
     return _upload(this.#client, this.#httpRequest, assetType, body, options)
@@ -87,7 +88,7 @@ export class AssetsClient {
    */
   upload(
     assetType: 'file',
-    body: File | Blob | Buffer | NodeJS.ReadableStream,
+    body: UploadBody,
     options?: UploadClientConfig
   ): Promise<SanityAssetDocument>
   /**
@@ -99,7 +100,7 @@ export class AssetsClient {
    */
   upload(
     assetType: 'image',
-    body: File | Blob | Buffer | NodeJS.ReadableStream,
+    body: UploadBody,
     options?: UploadClientConfig
   ): Promise<SanityImageAssetDocument>
   /**
@@ -111,12 +112,12 @@ export class AssetsClient {
    */
   upload(
     assetType: 'file' | 'image',
-    body: File | Blob | Buffer | NodeJS.ReadableStream,
+    body: UploadBody,
     options?: UploadClientConfig
   ): Promise<SanityAssetDocument | SanityImageAssetDocument>
   upload(
     assetType: 'file' | 'image',
-    body: File | Blob | Buffer | NodeJS.ReadableStream,
+    body: UploadBody,
     options?: UploadClientConfig
   ): Promise<SanityAssetDocument | SanityImageAssetDocument> {
     const observable = _upload(this.#client, this.#httpRequest, assetType, body, options)
@@ -137,7 +138,7 @@ function _upload(
   client: SanityClient | ObservableSanityClient,
   httpRequest: HttpRequest,
   assetType: 'image' | 'file',
-  body: File | Blob | Buffer | NodeJS.ReadableStream,
+  body: UploadBody,
   opts: UploadClientConfig = {}
 ): Observable<HttpRequestEvent<{document: SanityAssetDocument | SanityImageAssetDocument}>> {
   validators.validateAssetType(assetType)

--- a/src/types.ts
+++ b/src/types.ts
@@ -1,3 +1,4 @@
+// deno-lint-ignore-file no-empty-interface
 import type {Requester} from 'get-it'
 
 /**
@@ -5,6 +6,14 @@ import type {Requester} from 'get-it'
  * @internal
  */
 export type Any = any // eslint-disable-line @typescript-eslint/no-explicit-any
+
+declare global {
+  // Declare empty stub interfaces for environments where "dom" lib is not included
+  interface File {}
+}
+
+/** @public */
+export type UploadBody = File | Blob | Buffer | NodeJS.ReadableStream
 
 /** @public */
 export interface RequestOptions {

--- a/test/warnings.test.ts
+++ b/test/warnings.test.ts
@@ -19,7 +19,7 @@ describe('Client config warnings', async () => {
     )
   })
 
-  test.skipIf(isEdge)('warns if in browser on localhost and a token is provided', () => {
+  test('warns if in browser on localhost and a token is provided', () => {
     const restoreWindow = global.window
     global.window = {location: {hostname: 'localhost'}} as any
     createClient({projectId: 'abc123', useCdn: false, token: 'foo', apiVersion: '1'})

